lib/templates.php

```diff
@@ -155,3 +155,55 @@ function gutenberg_render_template_list_table_column( $column_name, $post_id ) {
 	echo esc_html( $post->post_name );
 }
 add_action( 'manage_wp_template_posts_custom_column', 'gutenberg_render_template_list_table_column', 10, 2 );
+
+/**
+ * Filter for adding a `resolved` parameter to `wp_template` queries.
+ *
+ * @param array $query_params The query parameters.
+ * @return array Filtered $query_params.
+ */
+function filter_rest_wp_template_collection_params( $query_params ) {
+	$query_params += array(
+		'resolved' => array(
+			'description' => __( 'Whether to filter for resolved templates', 'gutenberg' ),
+			'type'        => 'boolean',
+		),
+	);
+	return $query_params;
+}
+apply_filters( 'rest_wp_template_collection_params', 'filter_rest_wp_template_collection_params', 99, 1 );
+
+/**
+ * Filter for supporting the `resolved` parameter in `wp_template` queries.
+ *
+ * @param array           $args    The query arguments.
+ * @param WP_REST_Request $request The request object.
+ * @return array Filtered $args.
+ */
+function filter_rest_wp_template_query( $args, $request ) {
+	if ( $request['resolved'] ) {
+		$template_ids  = array( 0 ); // Return nothing by default (the 0 is needed for `post__in`).
+		$template_type = $request['slug'] ? array( $request['slug'] ) : get_template_types();
```

Comment:
This gets shadowed before it's used.

Reply:
Uh, half-reverted refactor. Good spot, I'll fix.

Reply:
7d72ff7